subprojects/core/src/main/java/org/gradle/api/internal/changedetection/TaskArtifactState.java

```diff
@@ -41,10 +41,10 @@
     boolean isAllowedToUseCachedResults();
 
     /**
-     * Returns the calculated cache key for the task's current state, or {@code null} if the task is not cacheable.
+     * Initializes and returns the current execution.
      */
     @Nullable
-    BuildCacheKey calculateCacheKey();
+    TaskExecution getCurrentExecution();
```

Comment:
Can we keep `TaskExecution` hidden and just let this be a pass-through to get the `BuildCacheKey`?